### PR TITLE
Adds the roach language and a way to speak it

### PR DESCRIPTION
**Pull request by Alliostra**
## About The Pull Request

Adds a language spoken by roaches. You can learn it by crafting a makeshift language imprinter from some electronic parts and some fuhrer meat.
Taming roaches is now affected by both your COG stat, and whether or not you speak the roach language. Vagabonds have a small chance to spawn with knowledge of how to speak the roach language

## Why It's Good For The Game

You can now talk ~~turkish~~ to mind swapped roachies! rejoice!

## Changelog
:cl:
add: Adds the roach language and a crafted roach language imprinter
tweak: COG and wheter or not you speak the roach language now affects roach taming success rate.
tweak: vagabonds now have a chance to know the roach language from the get-go
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/language_imprinter.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/obj/item/device/language_imprinter/attack(mob/M, mob/living/carbon/human/user, target_zone)
+	if(!istype(user) || user.incapacitated() || spent)
+		return ..()
+	if(length(M.get_covering_equipped_items(EYES)))
+		to_chat(user, SPAN_WARNING("You need to remove the eye covering first."))
+		return ..()
+	M.add_language(language)
+
+	to_chat(user, SPAN_DANGER("[src] plunges into [M]'s eye!"))
```
Comment (Firefox13) [inline code/game/objects/items/devices/language_imprinter.dm:18]: If you're copying the mental imprintor's flavour, add the sanity damage too.

File: code/game/objects/items/devices/language_imprinter.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/obj/item/device/language_imprinter
+	name = "language imprinter"
+	desc = "A device that is applied to an eye to imprint the sol common language to one's mind."
```
Comment (Firefox13) [inline code/game/objects/items/devices/language_imprinter.dm:3]:
```suggestion
	desc = "A device that is applied to an eye to imprint the English Common language to one's mind."
```

**Review by Firefox13 — CHANGES_REQUESTED** [review]

**Comment by Alanii** [comment]
What are the verbages for the roachies?

This is very neat.

**Comment by Alliostra** [comment]
If you don't understand the language, you just hear "John McVagabond chirrups/chirps/clicks"

File: code/game/objects/items/devices/language_imprinter.dm (hunk 1 above)
Comment (AlexMorgan3817) [reply code/game/objects/items/devices/language_imprinter.dm:20]: Do not copy, use inheriting

**Pull request closed without merging**